stravalib/model.py

```diff
@@ -373,32 +588,66 @@ class Athlete(
     membership: Optional[str] = None
     admin: Optional[bool] = None
     owner: Optional[bool] = None
-    subscription_permissions: Optional[list] = None
+    # TODO: can permission take other inputs other than bool?
+    subscription_permissions: Optional[List[bool]] = None
 
     @validator("athlete_type")
-    def to_str_representation(cls, raw_type):
-        # Replaces legacy "ChoicesAttribute" class
+    def to_str_representation(cls, raw_type: int) -> Optional[str]:
+        """Replaces legacy 'ChoicesAttribute' class.
+
+        Parameters
+        ----------
+        raw_type : int
+            The raw integer representing the athlete type.
+
+        Returns
+        -------
+        str
+            The string representation of the athlete type.
+        """
+
+        # return type "str"
         return {0: "cyclist", 1: "runner"}.get(raw_type)
 
     @lazy_property
-    def authenticated_athlete(self):
+    def authenticated_athlete(self) -> Athlete:
+        """
+        Returns an `stravalib.Client` object containing Strava account data
+        for the (authenticated) athlete.
+
+        Returns
+        -------
+        DetailedAthlete
+            The detailed information of the authenticated athlete.
+        """
+        assert self.bound_client is not None, "Bound client is not set."
```

Comment:
mypy wants us to ensure bound_client is not None through out either assert of a conditional work. i went with asserts as it feels cleaner. i am not actually sure how the client could be None to begin with TBH. but the check is easy to add.

Reply:
Yeah if there's no good reason for the value to be None I think it's better going with an assert

Reply:
ok great - i just am not sure why `to_str_representation ` above isn't fixed with an assert. i'll resolve this one given it sounds like my approach is generally good here!!